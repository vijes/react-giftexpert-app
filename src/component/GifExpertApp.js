import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    // const categories = ['One punch', 'Samurai X', 'Dragon Ball'];

    // const [categories, setCategories] = useState(['Samurai X']);
    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = () => {
    //     setCategories( ['Dragon Ball GT', ...categories] );
    //     // setCategories( cats => [ ... cats, 'Dagon Ball GT']);
    // }

    return (
        <>
            <h2>GiftExpertApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />

            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                            key={ category }
                            category={ category } 
                        />
                    ))
                }
            </ol>
        </>
    );
}

export default GifExpertApp;