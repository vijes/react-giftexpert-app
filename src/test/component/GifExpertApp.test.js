import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../../component/GifExpertApp';

describe('Pruebas con el <GifExpertApp />', () => {

    test('Consumir el componente <GifExpertApp /> correctamente', () => {

        const wrapper = shallow( <GifExpertApp />);

        expect( wrapper ).toMatchSnapshot();
        
    });
    
    test('debe de mostrar una lista de categorias', () => {

        const categories = ['Dragon ball', 'Dragon Ball Z'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );        
    });
    
    
});