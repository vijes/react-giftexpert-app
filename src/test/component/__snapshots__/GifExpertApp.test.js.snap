// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el <GifExpertApp /> Consumir el componente <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas con el <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
  </ol>
</Fragment>
`;
